pkg/stub/handler.go

```diff
@@ -2,96 +2,410 @@ package stub
+
+	h.initter = &defaultInClusterInitter{h: &h}
+	h.initter.init()
+
+	h.sdkwrapper = &defaultSDKWrapper{h: &h}
+
+	h.fileimagegetter = &defaultImageStreamFromFileGetter{h: &h}
+	h.filetemplategetter = &defaultTemplateFromFileGetter{h: &h}
+	h.filefinder = &defaultResourceFileLister{h: &h}
+
+	h.imageclientwrapper = &defaultImageStreamClientWrapper{h: &h}
+	h.templateclientwrapper = &defaultTemplateClientWrapper{h: &h}
+	h.secreteclientwrapper = &defaultSecretClientWrapper{h: &h}
+
+	h.namespace = getNamespace()
+
+	h.skippedImagestreams = make(map[string]bool)
+	h.skippedTemplates = make(map[string]bool)
+
+	return &h
 }
 
 type Handler struct {
-	// Fill me
+	initter InClusterInitter
+
+	sdkwrapper SDKWrapper
+
+	operatorCfg    *v1alpha1.SamplesResource
+	registrySecret *corev1.Secret
+
+	restconfig  *restclient.Config
+	tempclient  *templatev1client.TemplateV1Client
+	imageclient *imagev1client.ImageV1Client
+	coreclient  *corev1client.CoreV1Client
+
+	imageclientwrapper    ImageStreamClientWrapper
+	templateclientwrapper TemplateClientWrapper
+	secreteclientwrapper  SecretClientWrapper
+
+	fileimagegetter    ImageStreamFromFileGetter
+	filetemplategetter TemplateFromFileGetter
+	filefinder         ResourceFileLister
+
+	namespace string
+
+	skippedTemplates    map[string]bool
+	skippedImagestreams map[string]bool
+}
+
+func (h *Handler) manageDockerCfgSecret(deleted bool) error {
+	err := h.secreteclientwrapper.Delete("openshift", h.registrySecret.Name, &metav1.DeleteOptions{})
+	if err != nil && !kerrors.IsNotFound(err) {
+		return h.processError(v1alpha1.SecretUpdateFailed, err, "failed to delete before create dockerconfig secret the openshift namespace: %v")
+	}
+	if deleted {
+		logrus.Printf("registry dockerconfig secret %s was deleted", h.registrySecret.Name)
+		h.registrySecret = nil
+		h.operatorCfg.Status.SecretCreated = false
+		h.sdkwrapper.Update()
+	} else {
+		secretToCreate := corev1.Secret{}
+		h.registrySecret.DeepCopyInto(&secretToCreate)
+		secretToCreate.Namespace = ""
+		secretToCreate.ResourceVersion = ""
+
+		_, err = h.secreteclientwrapper.Create("openshift", &secretToCreate)
+		if err != nil {
+			return h.processError(v1alpha1.SecretUpdateFailed, err, "failed to create registry dockerconfig secret in openshift namespace : %v")
+		}
+		logrus.Printf("created/updated dockerconfig secret %s in openshift namespace", h.operatorCfg.Spec.RegistryCredentials.Name)
+
+		h.operatorCfg.Status.SecretCreated = true
+		h.sdkwrapper.Update()
+
+	}
+
+	status := v1alpha1.ResourceCondition{
+		Type: v1alpha1.SecretUpdateComplete,
+		Time: kapis.Now(),
+	}
+	h.operatorCfg.Status.Conditions = append(h.operatorCfg.Status.Conditions, status)
+	err = h.sdkwrapper.Update()
+	if err != nil {
+		if !kerrors.IsConflict(err) {
+			return h.processError(v1alpha1.SamplesUpdateFailed, err, "failed adding success condition to config: %v")
+		}
+		logrus.Println("got conflict error on success status update, just going to forgo this time vs. retry")
```

Comment:
// TODO - should requeue this event w/ a backoff for retry.

Reply:
Ok, in operator-sdk parlance, that would be returning an error.  The sdk's informer-sync.go error handling does the requeue, and employs rate limiting, using the k8s client-go APIs.

change to return the error part of the next push ... still going to distinguish conflict and non-conflict, so I can log the weird errors separately

Reply:
the sdk does not expose a way for me to initiate requeue etc.

Reply:
id *does* result in a requeue with backoff

Reply:
ok, i was confused by "the sdk does not expose a way for me to initiate requeue etc."